master/buildbot/worker/kubernetes.py

```diff
@@ -0,0 +1,368 @@
+                            config, self.config_loader_list
+                        )
+                    )
+            if not config_loader:
+                continue
+            try:
+                return config_loader()
+            except KubeClientConfigException:
+                continue
+        else:
+            raise KubeClientConfigException(
+                'Cannot find a working configurator'
+            )
+
+    def stopService(self):
+        if self._cert_file:
+            os.remove(self._cert_file)
+        return HTTPClientService.stopService(self)
+
+    def _prepareRequest(self, ep, kwargs):
+        url, req_kwargs = HTTPClientService._prepareRequest(self, ep, kwargs)
+        if self._cert is not None and 'cert' not in req_kwargs:
+            req_kwargs['cert'] = self._cert
+        if self._verify is not None and 'verify' not in req_kwargs:
+            req_kwargs['verify'] = self._verify
+        return url, req_kwargs
+
+    @property
+    def kube_config_loader(self):
+        if HAS_YAML:
+            return self._kube_config_loader
+        else:
+            log.showwarning(
+                "python-yaml module is not available, "
+                "can't load kubeconfig file."
+            )
+            return None
+
+    def _kube_config_loader(self):
+        kube_config = os.path.expanduser(self.kube_config_file)
+        try:
+            with open(kube_config) as yaml_config:
+                config_tree = yaml.load(yaml_config)
+        except FileNotFoundError:
+            raise KubeClientConfigException('Kubeconfig file not found')
+        try:
+            current_context_name = config_tree['current-context']
+        except KeyError:
+            raise ValueError("Cannot find 'current-context'")
+        context = self.search_section(
+            config_tree['contexts'],
+            current_context_name
+        )
+        cluster_name = context['context']['cluster']
+
+        cluster = self.search_section(config_tree['clusters'], cluster_name)
+        request_kwargs = {}
+        url = self.load_server_config(cluster['cluster'], request_kwargs)
+        user_name = context['context']['user']
+        user = self.search_section(config_tree['users'], user_name)
+        self.load_user_config(user['user'], request_kwargs)
+        namespace = context['context'].get('namespace', 'default')
+        return url, request_kwargs, namespace
+
+    def load_server_config(self, cluster, request_kwargs):
+        try:
+            url = cluster['server']
+        except KeyError:
+            raise ValueError(
+                "cluster config misses mandatory value: 'cluster.server'"
+            )
+        skip_ssl = cluster.get('insecure-skip-tls-verify', False)
+        cert_data = cluster.get('certificate-authority-data', None)
+        cert_path = cluster.get('certificate-authority', None)
+        if cert_data:
+            tmp = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
I would put that file in ~/.kube/<certsha>.crt and don't bother with deleting it.